src/Adapter/Order/CommandHandler/AddOrderFromBackOfficeHandler.php

```diff
@@ -84,14 +79,8 @@ public function handle(AddOrderFromBackOfficeCommand $command)
         $this->assertAddressesAreNotDisabled($cart);
 
         //Context country, language and currency is used in PaymentModule::validateOrder (it should rely on cart address country instead)
-        $this->contextStateManager
-            ->setCart($cart)
-            ->setCurrency(new Currency($cart->id_currency))
-            ->setCustomer(new Customer($cart->id_customer))
-            ->setLanguage(new Language($cart->id_lang))
-            ->setCountry($this->getTaxCountry($cart))
-            ->setShop(new Shop($cart->id_shop))
-        ;
+        $this->setCartContext($this->contextStateManager, $cart);
```

Comment:
`setCartContext` is triggering `saveCurrentContext()` at the beginning just to have:
`$this->contextStateManager->saveCurrentContext();`

right after it? 🤔 is this for purpose?

Reply:
You're right, there is an extra `saveCurrentContext` in this class that should be removed, `saveCurrentContext` should always be called BEFORE you override the context, so its rightful place is in the `setCartContext` function

Reply:
Fixed